apps/core0/subsys/kvm/manager.go

```
@@ -496,6 +509,40 @@ func (m *kvmManager) getDomainStruct(uuid string) (*Domain, error) {
 	if err != nil {
 		return nil, fmt.Errorf("cannot parse the domain xml: %v", err)
 	}
+
+	domainMetaData, err := domain.GetMetadata(libvirt.DOMAIN_METADATA_ELEMENT, metadataUri, libvirt.DOMAIN_AFFECT_LIVE)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get tags metadata for domain with the uuid %s", uuid)
```

why u hiding the original error? it's either u return the err directly, or at least add it to your customer error. But now, i have no idea why it couldn't get tags metadata.